src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileFormat.cs

```diff
@@ -69,6 +71,30 @@ public LockFile Parse(string lockFileContent, ILogger log, string path)
             }
         }
 
+        public LockFile SafeRead(string filePath, ILogger log)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+
+            var retries = 3;
+            for (var i = 1; i <= retries; i++)
+            {
+                // Ignore exceptions for the first attempts
+                try
+                {
+                    return Read(filePath, log);
```

Comment:
Consider making a helper method for this in FileUtility, you can see other examples there of how a Func can be passed that does the actual read and then returns a generic. Not critical for this PR however, I could help you with it afterwards.